src/functions/variables/getLeaderboardInfo.js

```diff
@@ -14,39 +23,33 @@ module.exports = async (d) => {
     key = `${variable.addBrackets()}_${`${id}_${
       d.guild?.id === undefined ? "dm" : d.guild?.id
     }`}`;
-    cache = await d.client.guilds.cache.get(id);
+    cache = await d.util.getGuild(d, id);
     user =
-      typeof cache === "undefined" || Object.keys(cache).length === 0
-        ? await d.client.users.fetch(id)
+      typeof cache === undefined || Object.keys(cache).length === 0
+        ? await d.util.getUser(d, id)
         : undefined;
   } else if (type === "global") {
     key = `${variable.addBrackets()}_${id}`;
-    cache = await d.client.guilds.cache.get(id);
+    cache = await d.util.getGuild(d, id);
     user =
-      typeof cache === "undefined" || Object.keys(cache).length === 0
-        ? await d.client.users.fetch(id)
+      typeof cache === undefined || Object.keys(cache).length === 0
+        ? await d.util.getUser(d, id)
         : undefined;
   } else if (type === "message" || type === "channel") {
     key = `${variable.addBrackets()}_${id}`;
   } else {
     d.aoiError.fnError(d, "custom", { inside: data.inside }, `type`);
   }
 
-  const foundData = all.find((x) => x.key === key);
-  const foundDataMap = all
-    .map((x, pos) => ({ value: x.value, key: x.key, pos: pos + 1 }))
-    .sort((a, b) => a.value - b.value);
-  let value = foundData?.value ?? 0;
-
   switch (format) {
     case "top":
-      data.result = foundDataMap.find((x) => x.key === key)?.pos || null;
+      data.result = all.slice().sort((a, b) => b.value - a.value).findIndex((x) => x.key === key) + 1 || 0;
       break;
     case "value":
-      data.result = value;
+      data.result = all.find(x => x.key === key).value || 0;
       break;
-    case "tag":
```

Comment:
Other leaderboard function support tag.
All should be same

Reply:
tag is going to be removed by discord though, returning null or 0000 wouldn make sense

Reply:
I can add tag but meh